app/javascript/lib/validation/validate-file-content.js

```diff
@@ -261,6 +275,104 @@ function validateNoMetadataCoordinates(parsedHeaders) {
+ * The main circumstance this is aimed at checking is the 'Excel drag error', in which by drag-copying a row, the
+ * label is copied correctly, but the id string gets numerically incremented
+ */
+function validateMetadataLabelMatches(parsedHeaders, parsedLine, isLastLine, dataObj) {
+  const issues = []
+  const excludedColumns = ['NAME']
+  // if this is the first time through, identify the colums to check, and initialize data structures to track mismatches
+  if (!dataObj.dragCheckColumns) {
+    dataObj.dragCheckColumns = parsedHeaders[0].map((colName, index) => {
+      const labelColumnIndex = parsedHeaders[0].indexOf(`${colName }__ontology_label`)
+      if (excludedColumns.includes(colName) ||
+        colName.endsWith('ontology_label') ||
+        parsedHeaders[1][index] === 'numeric' ||
+        labelColumnIndex === -1) {
+        return null
+      }
+      // for each column, track a hash of label=>value,
+      // and also a set of mismatched values--where the same label is used for different ids
+      return { colName, index, labelColumnIndex, labelValueMap: {}, mismatchedVals: new Set() }
+    }).filter(c => c)
+  }
+  // for each column we need to check, see if there is a corresponding filled-in label,
+  //  and track whether other ids have been assigned to that label too
+  dataObj.dragCheckColumns.forEach(dcc => {
+    const colValue = parsedLine[dcc.index]
+    const label = parsedLine[dcc.labelColumnIndex]
+    if (label.length) {
+      if (dcc.labelValueMap[label] && dcc.labelValueMap[label] !== colValue) {
+        dcc.mismatchedVals.add(label)
+      } else {
+        dcc.labelValueMap[label] = colValue
+      }
+    }
+  })
+
+  // only report out errors if this is the last line of the file so that a single, consolidated message can be displayed per column
+  if (isLastLine) {
+    dataObj.dragCheckColumns.forEach(dcc => {
+      if (dcc.mismatchedVals.size > 0) {
+        const labelString = [...dcc.mismatchedVals].slice(0, 10).join(', ')
+        issues.push(['error', 'content:metadata:mismatched-id-label',
+          `${dcc.colName} has different id values mapped to the same label.
+          Label(s) with more than one corresponding id: ${labelString}`])
+      }
+    })
+  }
+  return issues
+}
+/** raises a warning if a group column has more than 200 unique values */
+function validateGroupColumnCounts(parsedHeaders, parsedLine, isLastLine, dataObj) {
+  const issues = []
+  const excludedColumns = ['NAME']
+  if (!dataObj.groupCheckColumns) {
+    dataObj.groupCheckColumns = parsedHeaders[0].map((colName, index) => {
+      if (excludedColumns.includes(colName) || colName.endsWith('ontology_label') || parsedHeaders[1][index] === 'numeric') {
+        return null
+      }
+      return { colName, index, uniqueVals: new Set() }
+    }).filter(c => c)
+  }
+
+  dataObj.groupCheckColumns.forEach(gcc => {
+    const colValue = parsedLine[gcc.index]
```

Comment:
For easier readability, following my comments in the upstream PR:

```suggestion
function validateGroupColumnCounts(headers, line, isLastLine, dataObj) {
  const issues = []
  const excludedColumns = ['NAME']
  if (!dataObj.groupCheckColumns) {
    dataObj.groupCheckColumns = headers[0].map((colName, index) => {
      if (excludedColumns.includes(colName) || colName.endsWith('ontology_label') || headers[1][index] === 'numeric') {
        return null
      }
      return { colName, index, uniqueVals: new Set() }
    }).filter(c => c)
  }

  dataObj.groupCheckColumns.forEach(gcc => {
    const colValue = line[gcc.index]
```